Add 3 labels to all resources, add e2e test that validate the existence of the three labels

Shai1-Levi opened the pull request:
#### Why we need this PR
To meet k8s custom labels standardization:
Add 3 custom labels to all resources, add e2e that test the existence of name label 

#### Changes made
1. Add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

2. Add test that validate retrieval of operator controller pod by the name label.

3. Add the default container label, for easier getting desired logs.


#### Which issue(s) this PR fixes


#### Test plan

Commit:
1. Add custom label, propagate the label to all the right files.
2. Add test that validate retrieval of operator controller pod by the label.

Commit:
1. Add custom label, propagate the label to all the right files.
2. Add test that validate retrieval of operator controller pod by the label.

slintes commented:
in the past at least, label additions broke updates... [1]
IIRC e2e has an update test, let's see what happens.

/test 4.17-openshift-e2e

[1] https://github.com/medik8s/node-healthcheck-operator/commit/df1d2e8f1463fc94a77d11cb99c99ee8014d5006#diff-c006b1ef995eeeb57408cce42bd84db8dfc0cbe18df03222857665cebb1ac6fdR9-R18

slintes commented:
update still fails 🤷🏼‍♂️ 

`"Failed to run bundle upgrade: error waiting for CSV to install: csv failed: reason: \"InstallComponentFailed\", message: \"install strategy failed: Deployment.apps \\\"node-maintenance-operator-controller-manager\\\" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{\\\"app.kubernetes.io/name\\\":\\\"node-maintenance-operator\\\", \\\"control-plane\\\":\\\"controller-manager\\\", \\\"node-maintenance-operator\\\":\\\"\\\"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable\"\n"
make: *** [Makefile:208: bundle-run-update] Error 1`

Commit:
Fix labels:
- add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

- Add test that validate retrieval of operator controller pod by the label.

Shai1-Levi commented:
/test 4.17-openshift-e2e

Commit:
Fix labels:
- add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

- Add test that validate retrieval of operator controller pod by the label.

slintes commented:
/test 4.17-openshift-e2e

slintes commented:
operator update succeeded
test failure looks unrelated, looking at the test history this seems to be very flaky... :/ 

/ok-to-test
/lgtm
/approve

Commit:
Fix labels:
- Add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

- Add test that validate retrieval of operator controller pod by the label.

- Add the default container label, for easier getting desired logs.

Shai1-Levi commented:
Hey,
I added some labels to fix the e2e test and test-scorecard 
@slintes I didn't saw your last comment before I pushed the code

/test 4.17-openshift-e2e

slintes commented on bundle/manifests/node-maintenance-operator.clusterserviceversion.yaml:
```diff
@@ -18,8 +18,8 @@ metadata:
     categories: OpenShift Optional
-    containerImage: ""
-    createdAt: ""
+    containerImage: quay.io/rh-ee-slevi/node-maintenance-operator:latest
```
unwanted change

Shai1-Levi replied:
Thanks

slintes commented on bundle/manifests/node-maintenance-operator.clusterserviceversion.yaml:
```diff
@@ -18,8 +18,8 @@ metadata:
-    containerImage: ""
-    createdAt: ""
+    containerImage: quay.io/rh-ee-slevi/node-maintenance-operator:latest
+    createdAt: 2025-01-16 19:42:40
```
unwanted change

Shai1-Levi replied:
Thanks

Commit:
Fix labels:
- Add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

- Add test that validate retrieval of operator controller pod by the label.

- Add the default container label, for easier getting desired logs.

slintes commented:
/retest
/lgtm

slintes commented:
/retest

slintes commented:
setup failed...

/retest

slintes commented:
@razo7 since when is the e2e test so flaky? :/

```  STEP: Verifying that event SucceedMaintenance was created for test-maintenance nm CR - /go/src/github.com/medik8s/node-maintenance-operator/test/e2e/node_maintenance_test.go:497 @ 01/18/25 09:13:32.24
Error waiting for events: context deadline exceeded  [FAILED] Event SucceedMaintenance was missing for test-maintenance nm CR
  Unexpected error:
      <context.deadlineExceededError>: 
      context deadline exceeded
      {}
  occurred
```

slintes commented:
/retest

razo7 commented on test/e2e/node_maintenance_test.go:
```diff
@@ -419,6 +422,13 @@ func getOperatorPod() *corev1.Pod {
 }
 
+func getOperatorLabel() bool {
+	pods, err := KubeClient.CoreV1().Pods(operatorNsName).List(context.Background(), metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=node-maintenance-operator"})
```
IIUC the PR introduces 3 new labels while this function looks for the existence of only one of them. See also [K8s recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) .

Shai1-Levi replied:
Thanks, I will add them in another PR. 
Regard [k8s recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) I over this doc and I think that all the relevant labels was added. Thank you for introducing me with these k8s convention

Commit:
Add 3 custom labels to all resources, add e2e test that validate the labels existence
- Add recommended app.kubernetes.io/name label to everything

  With these changes we prevent services selecting pods by other
  operators, because they all have the control-plane: controller-manager
  label by default.

  Because deployment updates with modified labels fail, also rename the
  deployment by modifying the namePrefix (causes bundle file renames as
  well).

- Add tests that validate retrieval of operator controller pod by label.

- Add the default container label, for easier getting desired logs.

slintes reviewed (approved):
/retest

slintes commented:
/hold

since this is not urgent, let's first try to fix the e2e test, these retests are a waste of resources...

Commit:
Refactor tests

slintes commented on test/e2e/node_maintenance_test.go:
```diff
@@ -179,6 +179,9 @@ var _ = Describe("Starting Maintenance", func() {
 			Expect(getOperatorLogs()).To(ContainSubstring(nodemaintenance.FixedDurationReconcileLog))
 
+			//validate that operator controller pod have app.kubernetes.io/name label
+			Expect(validateOperatorCustomLabels()).To(BeTrue(), "operator custom label validation failed")
```
the return value and checking it isn't needed IMHO, everything is checked inside the function already?

slintes commented on test/e2e/node_maintenance_test.go:
```diff
@@ -419,6 +422,31 @@ func getOperatorPod() *corev1.Pod {
+	podName := podOperator.ObjectMeta.Name
+
+	// Get the Pod
+	pod, err := KubeClient.CoreV1().Pods(operatorNsName).Get(context.TODO(), podName, metav1.GetOptions{})
```
you have the pod already, why getting it here again?

slintes commented on test/e2e/node_maintenance_test.go:
```diff
@@ -419,6 +422,31 @@ func getOperatorPod() *corev1.Pod {
+	// Check if each label is present
+	for _, key := range requiredKeys {
+		_, exists := pod.Labels[key]
+		ExpectWithOffset(1, exists).Should(BeTrue(), fmt.Sprintf("Missing required label '%s' in pod '%s'", key, podName))
```
a) you don't check label values anymore
b) hint: ExpectWithOffset(1, pod.Labels).To(HaveKeyWithValue(key, value))
